cobbler/actions/importer.py

```diff
@@ -0,0 +1,162 @@
+"""
+This module contains the logic that kicks of the ``cobbler import`` process. This is extracted logic from ``api.py``
+that is essentially calling ``modules/mangers/import_signatures.py`` with some preparatory code.
+"""
+import logging
+import os
+
+from cobbler import utils
+from cobbler.utils import filesystem_helpers
+
+
+class Importer:
+    """
+    Wrapper class to adhere to the style of all other actions.
+    """
+
+    def __init__(self, api):
+        """
+        Constructor to initialize the class.
+
+        :param api: The CobblerAPI.
+        """
+        self.api = api
+        self.logger = logging.getLogger()
+
+    def run(
+        self,
+        mirror_url: str,
+        mirror_name: str,
+        network_root=None,
+        autoinstall_file=None,
+        rsync_flags=None,
+        arch=None,
+        breed=None,
+        os_version=None,
+    ) -> bool:
+        """
+        Automatically import a directory tree full of distribution files.
+
+        :param mirror_url: Can be a string that represents a path, a user@host syntax for SSH, or an rsync:// address.
+                           If mirror_url is a filesystem path and mirroring is not desired, set network_root to
+                           something like "nfs://path/to/mirror_url/root"
+        :param mirror_name: The name of the mirror.
+        :param network_root: the remote path (nfs/http/ftp) for the distro files
+        :param autoinstall_file: user-specified response file, which will override the default
+        :param rsync_flags: Additional flags that will be passed to the rsync call that will sync everything to the
+                            Cobbler webroot.
+        :param arch: user-specified architecture
+        :param breed: user-specified breed
+        :param os_version: user-specified OS version
+        """
+        self.api.log(
+            "import_tree",
+            [mirror_url, mirror_name, network_root, autoinstall_file, rsync_flags],
+        )
+
+        # Both --path and --name are required arguments.
+        if mirror_url is None or not mirror_url:
+            self.logger.info("import failed.  no --path specified")
+            return False
+        if mirror_name is None or not mirror_name:
```

Comment:
nitpick: the first check is redundant.
```suggestion
        if not mirror_name:
```